### PR TITLE
Let the user know to install cargo fmt

### LINKED ISSUE #1131: "Reformat file with rustfmt" gives confusing error if rustfmt is not installed
I hadn't realized that `rustfmt` is a separate package that isn't included by default with Cargo, and I hadn't installed it.

Without `rustfmt` installed, IntelliJ-Rust gives this error message:

 > C:\Users\<me>\.cargo\bin\cargo.exe fmt -- --write-mode=overwrite --skip-children <my-project>/<my-file>.rs stdout : stderr : error: no such subcommand: `fmt` Did you mean `doc`?

No, I did not mean `doc`, silly tool!

Obviously this is user error, but I believe IntelliJ-Rust could probably handle this fairly easily, i.e. by checking for the above error and suggesting to the user to manually run `cargo install rustfmt`, or by simply automatically checking if `rustfmt` is installed and doing the install automatically if not.

### PR DESCRIPTION
**Pull request by maraisr**
Never written a single line of Kotlin before - so please be nice! :heart: 

fixes #1131

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/actions/RsFmtFileAction.kt [hunk 1]
```diff
@@ -48,7 +48,16 @@ class RsFmtFileAction : DumbAwareAction() {
         try {
             cargo.reformatFile(module, file)
         } catch (e: ExecutionException) {
-            project.showBalloon(e.message ?: "", NotificationType.ERROR)
+            // #1131 - Check if we get a `no such subcommand: fmt` and let the user know to install fmt
+            val messageCheck = Regex("no such subcommand: fmt");
```
Comment (matklad) [inline src/main/kotlin/org/rust/ide/actions/RsFmtFileAction.kt]: In Kotlin, there's a handy `toRegex` extension, so it's more idiomatic to write `"no such subcommand: fmt".toRegex()`.

However in this case the regex is trivial, so we can just search for a substring, using the `in` operator:

```
if ("no such subcommand: fmt" in e.message) {
    ...
}
```

**Comment by matklad** [comment]
Great, thanks @maraisr !

Please note that we require CLA's for pull request, as per our [CONTRIBUTING.md](https://github.com/intellij-rust/intellij-rust/blob/master/CONTRIBUTING.md#contributing):

>We require a contributor license agreement for all contributions. You can sign it electronically at https://www.jetbrains.com/agreements/cla/ via DocuSign. The procedure consists of a couple of emails and clicks: no need to scan or print or snail-mail anything :) CONTRIBUTORS.txt file stores GitHub usernames of people who already signed a CLA.

Feel free to add yourself to CONTRIBUTORS.txt after signing :)

File: src/main/kotlin/org/rust/ide/actions/RsFmtFileAction.kt [hunk 2]
```diff
@@ -48,7 +48,12 @@ class RsFmtFileAction : DumbAwareAction() {
         try {
             cargo.reformatFile(module, file)
         } catch (e: ExecutionException) {
-            project.showBalloon(e.message ?: "", NotificationType.ERROR)
+            // #1131 - Check if we get a `no such subcommand: fmt` and let the user know to install fmt
+            if ("no such subcommand: fmt" in e.message) {
```
Comment (matklad) [inline src/main/kotlin/org/rust/ide/actions/RsFmtFileAction.kt]: Hm, looks like `e.message` is nullable on at least one version of IJ. I think we can do

```
val message = e.message ?: ""
```

and then use it in this condition and also in the else branch.

Comment (maraisr) [reply src/main/kotlin/org/rust/ide/actions/RsFmtFileAction.kt]: I'll after this PR definitely go learn Kotlin! Silly mistakes like this shouldn't be happening.

**Comment by maraisr** [comment]
@matklad  thank you again so much for this!

**Comment by matklad** [comment]
This is already released: https://intellij-rust.github.io/2017/10/02/changelog-53.html :)

**Pull request merged**